### DISCUSSION TITLE
How to serve the /docs folder on a different port

### DISCUSSION THREAD

**Discussion by TedGrahamEmporia** [line/armeria#5521] [General]
I want my gRPC service exposed on a different port than the DocService so I can control access to the docs via a whitelist.  If I create a separate ServerBuilder than the service will be exposed on the documentation port which seems wrong.  Suggestions for how to modify the below?

![image](https://github.com/line/armeria/assets/85959097/8cb54e81-baa2-4d3d-bbfe-cca6fdff94ae)

**Comment by ikhoon** [comment]
You can create a new virtual host which is bound to a specific port and register `DocService` there. 
https://github.com/line/armeria/blob/bc2454ac5d1d075af1bf8245d90da404905a9c3d/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1588-L1598
